relay-server/src/services/processor.rs

```diff
@@ -863,6 +866,17 @@ impl<'a, Group> ProcessEnvelopeState<'a, Group> {
     fn remove_event(&mut self) {
         self.event = Annotated::empty();
     }
+
+    /// Returns `true` for managed relays if a feature is disabled.
+    ///
+    /// Some envelope items are dropped based on a feature flag,
+    /// but we want to forward them in proxy mode.
+    fn feature_disabled_by_upstream(&self, feature: Feature) -> bool {
+        match self.config.relay_mode() {
+            RelayMode::Proxy | RelayMode::Static | RelayMode::Capture => false,
```

Comment:
Is this correct for all feature flags?

If not: Possibly we make `project.has_feature()` require a `&Config` as argument?

Reply:
I think this logic only makes sense for feature flags that are used to filter data. I.e. it would not make sense to assume that a feature is enabled for something like `Feature::DoExpensiveAndDangerousExperimentalStuff` in proxies. Do you think I should just rename this function to `should_filter`?

Reply:
I like `should_filter`!